arrow-core/build.gradle

```diff
@@ -8,13 +8,13 @@ apply from: "$SUB_PROJECT"
 apply from: "$DOC_CREATION"
 
 dependencies {
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$KOTLIN_VERSION"
-    compile project(":arrow-annotations")
-    compile project(":arrow-core-data")
+    api "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$KOTLIN_VERSION"
+    api project(":arrow-annotations")
+    api project(":arrow-core-data")
```

Comment:
This might come handy to save the users from self-importing this themselves, although I'm wondering if [this guide in the arrow readme](https://github.com/arrow-kt/arrow/blame/master/README.md#L105) was then accurate to the previous state 🤔  

I think we should check that document and validate if it's correct (for example arrow-optics doesn't seem to expose core, yet core doesn't appear in the readme for  optics)